Enhancement/5692 vulnerability scanner documentation (#29506)
wazuh/wazuh: docs/ref/modules/vulnerability-scanner/events.md

# Events format 

> The incoming events from the agents must be parseable by the [flatbufer schemas](flatbuffers.md), otherwise it will trigger an exception and the event won't reach the vulnerability scanner. Below we detail the format for the different systems supported by the scanner.

Reviewer: ```suggestion
The incoming events from the agents must be parseable by the [flatbuffer schemas](flatbuffers.md), otherwise it will trigger an exception, and the event won't reach the vulnerability scanner. Below, we detail the format for the different systems supported by the scanner.
```

## Syscollector event types

> The vulnerability scanner as an event-driven module, reacts to the syscollector events sent by the agents. There are two types of events received by the scanner: synchronization and delta events. 

Reviewer: ```suggestion
The vulnerability scanner as an event-driven module, reacts to the Syscollector events sent by the agents. There are two types of events received by the scanner: synchronization and delta events. 
```

> Synchronization events are sent during the first scan of the Syscollector scan, or if a delta event fails. 
> Delta events are the ones sent from the second Syscollector scan. 
> 
> those event types provide the same information to the scanner, the detection is not affected by the type of the event sent, they are just parsed differently.

Reviewer: ```suggestion
- Synchronization events are sent during the first scan of the Syscollector scan or if a delta event fails. 
- Delta events are the ones sent from the second Syscollector scan. 

These event types provide the same information to the scanner, the detection is not affected by the type of event sent, they are just parsed differently.
```

> Within synchronization events, an **integrity_clear** event will remove the information for certain provider. For the case of package provider it will remove all vulnerabilities related to packages, for OS will remove vulnerabilities related to OS but also affect the detection for packages, since the source operating system is a required information for detection.

Reviewer: ```suggestion
Within synchronization events, an **integrity_clear** event will remove the information for a certain provider. For the case of a package provider, it will remove all vulnerabilities related to packages, for OS will remove vulnerabilities related to OS, but also affect the detection for packages, since the source operating system is a required information for detection.
```

In the case of hotfixes it should detect vulnerabilities fixed by the presence of those security patches.  

## Hotfix events (Only Windows)

### Delta event

```json
{
    "agent_info": {
        "agent_id": "002"
    },
    "data_type": "dbsync_hotfixes",
    "data": {
        "checksum": "1691178971959743855",
        "hotfix": "KB5034763",
        "scan_time": "2023/08/04 19:56:11"
    },
    "operation": "INSERTED"
}
```

### Synchronization event

```json
{
    "agent_info": {
        "agent_id": "002"
    },
    "data_type": "state",
    "data": {
        "attributes_type": "syscollector_hotfixes",
        "attributes": {
            "checksum": "1691178971959743855",
            "hotfix": "KB3114960",
            "scan_time": "2023/08/04 19:56:11"
        }
    }
}
```

- Integrity clear event
```json
{
    "agent_info": {
        "agent_id": "002"
    },
    "data_type": "integrity_clear",
    "data": {
        "id": 1700236640,
        "attributes_type": "syscollector_hotfixes"
    }
}
```

## Os events 

> For OS events we need to pay attention to the following fields

Reviewer: ```suggestion
For OS events, we need to pay attention to the following fields:
```

            "hostname": "archlinux",
            "architecture": "x86_64",
            "os_name": "Arch Linux",
            "os_build": "rolling",
            "os_platform": "arch",
            "sysname": "Linux",
            "release": "6.9.7-arch1-1",
            "version": "#1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000",
            "checksum": "1720120521357595973"
        }
    }
}
```

- MacOS
```json
{
  "agent_info": {
    "agent_id": "001"
  },
  "data_type": "state",
  "data": {
    "attributes_type": "syscollector_osinfo",
    "attributes": {
      "architecture":"x86_64",
      "checksum":"1691178971959743855",
      "hostname":"fd9b83c25f30",
      "os_major":"14",
      "os_minor":"0",
      "os_name":"macOS",
      "os_platform":"darwin",
      "os_version":"14.0",
      "release":"5.4.0-155-generic",
      "scan_time":"2023/08/04 19:56:11",
      "sysname":"macOS",
      "version":"darwin 23.0"
    }
  }
}
```

- Integrity clear event
```json
{
    "agent_info": {
        "agent_id": "002"
    },
    "data_type": "integrity_clear",
    "data": {
        "id": 1700236640,
        "attributes_type": "syscollector_osinfo"
    }
}
```

## Package events 

> For package events we need to pay attention to the following fields

Reviewer: ```suggestion
For package events, we need to pay attention to the following fields:
```

> - vendor: Informs the source of the vulnerability. Different vendors have different treatmens for their packages. Supported vendors

Reviewer: ```suggestion
- vendor: Informs the source of the vulnerability. Different vendors have different treatments for their packages. Supported vendors are:
```

```console
Canonical
Ubuntu
Debian
Red Hat, Inc.
CentOS
Amazon Linux
Amazon.com
Amazon AWS
Arch Linux
suse
SUSE
openSUSE
AlmaLinux
CloudLinux
Rocky
```

> - format: When the package is not installed through a package manager apt, yum, pacman, etc. The format field must be specified if we are referring to npm, pypi or snap packages.

Reviewer: ```suggestion
- format: When the package is not installed through a package manager (`apt`, `yum`, `pacman`, etc). The format field must be specified if we are referring to `npm`, `pypi` or `snap` packages.
```

> - name and version are the main identifier of the package.

Reviewer: ```suggestion
- name and version are the main identifiers of the package.
```

> - item_id: A non-ambiguos alpha-numeric identifier that would be required for indexing. 

Reviewer: ```suggestion
- item_id: A non-ambiguous alphanumeric identifier that would be required for indexing. 
```
